src/peripheral/scb.rs

```diff
@@ -640,4 +643,29 @@ impl SCB {
             (*Self::ptr()).icsr.write(SCB_ICSR_PENDSVCLR);
         }
     }
+
+    /// Set the PENDSTCLR bit in the ICSR register which will clear a pending SysTick interrupt
+    #[inline]
+    pub fn set_systick(&mut self) {
+        unsafe {
+            (*Self::ptr()).icsr.write(SCB_ICSR_PENDSTSET);
+        }
+    }
+
+    /// Check if PENDSTSET bit in the ICSR register is set meaning SysTick interrupt is pending
+    #[inline]
+    pub fn is_systick_pending() -> bool {
+        unsafe {
+            (*Self::ptr()).icsr.read() & SCB_ICSR_PENDSTSET == SCB_ICSR_PENDSTSET
+        }
+    }
+
+    
+    /// Set the PENDSTCLR bit in the ICSR register which will clear a pending SysTick interrupt
+    #[inline]
+    pub fn clear_systick_pending(&mut self) {
```

Comment:
Does this operation require exclusive access to SCB (`&mut self`)? With `clear_pendsv()` it wasn't required because setting just the `PENDSTCLR` bit doesn't affect any other bits in the register, so you can't have data races.

Reply:
Yes, exclusive access to SCB is not required. I think that `clear_systick_pending` needs to change
 analogiously.